lib/experimental/fonts-library/class-wp-font-family-utils.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Fonts Family Utils class.
+ *
+ * This file contains utils fot Font Family class.
+ *
+ * @package    Gutenberg
+ * @subpackage Fonts Library
+ * @since      X.X.X
+ */
+
+if ( class_exists( 'WP_Font_Family_Utils' ) ) {
+	return;
+}
+
+/**
+ * A class of utilities for working with the Font Library.
+ */
+class WP_Font_Family_Utils {
+
+	/**
+	 * Generates a filename for a font face asset.
+	 *
+	 * Creates a filename for a font face asset using font family, style, weight and extension information.
+	 *
+	 * @param string $font_slug The font slug to use in the filename.
+	 * @param array  $font_face The font face array containing 'fontFamily', 'fontStyle', and 'fontWeight' attributes.
+	 * @param string $url The URL of the font face asset, used to derive the file extension.
+	 * @param string $suffix Optional suffix added to the resulting filename
+	 * @return string The generated filename for the font face asset.
+	 */
+	public static function get_filename_from_font_face( $font_slug, $font_face, $url, $suffix = '' ) {
+		$extension = pathinfo( $url, PATHINFO_EXTENSION );
+		$filename  = "{$font_slug}_{$font_face['fontStyle']}_{$font_face['fontWeight']}";
+		if ( $suffix ) {
```

Comment:
Nitpick: I would suggest using a strict comparison to check if the string is empty.
`"0"` could also be a valid prefix, but it will be ignored in the current implementation.
```suggestion
		if ( '' !== $suffix ) {
```